apps/sms/views/shared/js/localization_helper.js

```diff
@@ -25,32 +23,29 @@
    * changed and we need to update only elements with time aware formats.
    */
   function localizeDateTime(isTimeFormatChanged) {
-    var dateTimeElements = document.querySelectorAll(isTimeFormatChanged ?
-      '.l10n-contains-date[data-l10n-date-format12][data-l10n-date-format24]' :
-      '.l10n-contains-date'
-    );
+    var dateTimeElements = document.querySelectorAll('.l10n-contains-date');
+
     Array.forEach(dateTimeElements, function(element) {
       if (!element.dataset.l10nDate) {
         return;
       }
 
-      // If element contains time-aware format then we should respect current
-      // time format setting
       var formatL10nId = element.dataset.l10nDateFormat;
-      if (element.dataset.l10nDateFormat12 &&
-          element.dataset.l10nDateFormat24) {
-        formatL10nId = navigator.mozHour12 ?
-          element.dataset.l10nDateFormat12 : element.dataset.l10nDateFormat24;
-      }
 
       if (!formatL10nId) {
         return;
       }
 
-      var localeData = Utils.date.format.localeFormat(
-        new Date(+element.dataset.l10nDate),
-        navigator.mozL10n.get(formatL10nId)
-      );
+      formatL10nId = JSON.parse(formatL10nId);
+      
+      if (isTimeFormatChanged && !formatL10nId.hour) {
+        return;
+      }
+      formatL10nId.hour12 = navigator.mozHour12;
```

Comment:
too early. Intl is regulated by ECMA 402. There are several things we'd like to add in edition 2, but we need to first get navigator.hour12 before we can argue for including it into ECMA.